okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RemotingOkHttpCall.java

```diff
@@ -166,14 +194,18 @@ public void onFailure(Call call, IOException exception) {
                         client.newCallWithMutableState(redirectedRequest, backoffStrategy, maxNumRelocations - 1);
                 log.debug("Rescheduling call after backoff", SafeArg.of("backoffMillis", backoff.get().toMillis()),
                         exception);
-                scheduleExecution(() -> retryCall.enqueue(callback), backoff.get());
+                scheduleExecution(() -> {
+                    listener.onDropped();
+                    retryCall.enqueue(callback);
+                }, backoff.get());
             }
 
             @Override
             public void onResponse(Call call, Response response) throws IOException {
                 // Relay successful responses
                 if (response.code() / 100 <= 2) {
                     callback.onResponse(call, response);
+                    listener.onSuccess();
```

Comment:
the construction here is very brittle because you need to trace that the Listener is released (onSuccess, onDropped) on all code paths.

Reply:
yeah, i'm gonna clean that up